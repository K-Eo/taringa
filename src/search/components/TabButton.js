import React from 'react'
import classNames from 'classnames'

const TabButton = ({ className, id, isActive, label, onClick, totalCount }) => {
  const handleOnClick = e => {
    if (e && e.preventDefault) {
      e.preventDefault()
    }

    onClick(id)
  }

  const classes = classNames('nav-item nav-link', className, {
    active: isActive,
  })

  return (
    <a className={classes} onClick={handleOnClick}>
      {label}
      {totalCount !== null && (
        <span className="small text-secondary ml-1">({totalCount})</span>
      )}
    </a>
  )
}

TabButton.defaultProps = {
  id: '#ID',
  isActive: false,
  label: '#LABEL',
  onClick: () => {},
  totalCount: null,
}

export default TabButton
