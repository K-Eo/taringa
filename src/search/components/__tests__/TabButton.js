import React from 'react'

import TabButton from '../TabButton'

describe('TabButton', () => {
  it('exists', () => {
    expect(TabButton).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<TabButton />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with additional classnames', () => {
    const wrapper = shallow(<TabButton className="foo" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders active class when is active', () => {
    const wrapper = shallow(<TabButton isActive />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders totalCount next to label', () => {
    const wrapper = shallow(<TabButton label="foo" totalCount={10} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles onClick', () => {
    const mock = jest.fn()
    const wrapper = shallow(<TabButton id="foo" onClick={mock} />)

    wrapper.find('a').simulate('click')

    expect(mock).toHaveBeenCalledWith('foo')
  })
})
