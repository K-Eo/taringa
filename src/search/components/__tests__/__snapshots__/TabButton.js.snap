// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton renders active class when is active 1`] = `
<a
  className="nav-item nav-link text-dark active"
  href="/"
  onClick={[Function]}
>
  #LABEL
</a>
`;

exports[`TabButton renders default 1`] = `
<a
  className="nav-item nav-link text-dark"
  href="/"
  onClick={[Function]}
>
  #LABEL
</a>
`;

exports[`TabButton renders totalCount next to label 1`] = `
<a
  className="nav-item nav-link text-dark"
  href="/"
  onClick={[Function]}
>
  foo
  <span
    className="small text-secondary ml-1"
  >
    (
    10
    )
  </span>
</a>
`;

exports[`TabButton renders with additional classnames 1`] = `
<a
  className="nav-item nav-link text-dark foo"
  href="/"
  onClick={[Function]}
>
  #LABEL
</a>
`;
